woodwork/type_sys/inference_functions.py

```diff
@@ -78,6 +83,20 @@ def boolean_nullable_func(series):
         series.dtype,
     ):
         return True
+    elif pdtypes.is_object_dtype(series.dtype):
+        series_no_null = series.dropna()
+        try:
+            series_no_null_unq = set(series_no_null)
+            if series_no_null_unq in [
+                {False, True},
+                {True},
+                {False},
+            ]:
+                return True
+        except TypeError as te:  # Necessary to check for non-hashable values because of object dtype consideration
+            te_msg = str(te)
+            if "unhashable type" in te_msg:
```

Comment:
Do we technically not need this? It seems like either way, if we hit TypeError, we return False